src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1576,8 +1576,58 @@ static bool processSubscriptions
       continue;
     }
 
-    /* Check 3: expression (georel, which also uses geometry and coords) */
-    // TBD (issue #1678)
+    /* Check 3: expression (georel, which also uses geometry and coords)
+     * This should be always the last check, as it is the most expensive one, given that it interacts with DB
+     * (Issue #2396 should solve that) */
+    if ((tSubP->expression.georel != "") && (tSubP->expression.coords != "") && (tSubP->expression.geometry != ""))
+    {
+      Scope        geoScope;
+      std::string  filterErr;
+
+      if (geoScope.fill("v2", tSubP->expression.geometry, tSubP->expression.coords, tSubP->expression.georel, &filterErr) != 0)
+      {
+        // This has been already checked at subscription creation/update parsing time. Thus, the code cannot reach
+        // this part.
+        //
+        // (Probably the whole if clause will disapear when the missing part of #1705 gets implemented,
+        // moving geo-suff strings to a filter object in TriggeredSubscription class
+
+        LM_E(("Runtime Error (code cannot reach this point, error: %s)", filterErr.c_str()));
+        continue;
+      }
+
+      BSONObj areaFilter;
+      if (!processAreaScopeV2(&geoScope, areaFilter))
+      {
+        // Error in processAreaScopeV2 is intepreted as no-match (conservative approach)
+        continue;
+      }
+
+      // Look in the database of and entity that maches the geo-filters. Note that this query doesn't
```

Comment:
Yes, it is 'an'. I tend to use 'and' in place where I want to use 'an'. Like nofity ;)